packages/wonder-blocks-modal/src/components/flexible-dialog.tsx

```diff
@@ -0,0 +1,225 @@
+ * for aria-labelledby on the dialog to the heading content or a div wrapper.
+ */
+type TitleAndAriaLabelledBy = {
+    title: React.ReactNode;
+    "aria-label"?: never;
+    "aria-labelledby"?: string;
+};
+/**
+ * The optional accessible name of the modal.
+ * This is useful when there is no main heading in the dialog. A visible
+ * heading with an id is preferred for automatic matching with aria-labelledby.
+ */
+type AriaLabelOnly = {
+    title?: never;
+    "aria-label": string;
+    "aria-labelledby"?: never;
+};
+/**
+ * The optional ID of the content describing this dialog, if applicable.
+ */
+type AriaLabelledByOnly = {
+    title?: never;
+    "aria-label"?: never;
+    "aria-labelledby": string;
+};
+
+/**
+ * One of the labeling methods is required:
+ */
+type AccessibleLabelProps =
+    | TitleAndAriaLabelledBy
+    | AriaLabelOnly
+    | AriaLabelledByOnly;
+
+type Props = AccessibleLabelProps & {
+    /**
+     * The main heading of the modal, for labeling the dialog.
+     * If string content, an ID will be generated for the heading for aria-labelledby on the dialog.
+     * If a node, an ID will be applied to a DIV wrapping the node.
+     */
+    title?: React.ReactNode | string;
+    /**
+     * An optional id parameter for the main heading. If one is not provided,
+     * an ID will be generated.
+     */
+    titleId?: string;
+    /**
+     * The content of the modal. Supports a render prop for placing the title in a slot.
+     */
+    content:
+        | React.ReactElement<PropsFor<typeof FlexiblePanel>>
+        | ((slots: RenderProps) => React.ReactNode);
+    /**
+     * The background styles for the modal panel.
+     * If not provided, defaults to semanticColor.surface.primary background color.
+     */
+    backgroundStyles?: BackgroundStyles;
+    /**
+     * Called when the close button is clicked.
+     *
+     * If you're using `ModalLauncher`, you probably shouldn't use this prop!
+     * Instead, to listen for when the modal closes, add an `onClose` handler
+     * to the `ModalLauncher`.  Doing so will result in a console.warn().
+     */
+    onClose?: () => unknown;
+    /**
+     * When true, the close button is shown; otherwise, the close button is not shown.
+     */
+    closeButtonVisible?: boolean;
+    /**
+     * When set, provides a component that can render content above the top of the modal;
+     * when not set, no additional content is shown above the modal.
+     * This prop is passed down to the ModalDialog.
+     */
+    above?: React.ReactNode;
+    /**
+     * When set, provides a component that will render content below the bottom of the modal;
+     * when not set, no additional content is shown below the modal.
+     * This prop is passed down to the ModalDialog.
+     *
+     * NOTE: Devs can customize this content by rendering the component assigned to this prop with custom styles,
+     * such as by wrapping it in a View.
+     */
+    below?: React.ReactNode;
```

Comment:
Is there a use case for supporting `above` and `below` props in `FlexibleDialog`? I'm wondering if consumers need to use those props, should they use the `ModalDialog` kind instead? What do you think?